app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -94,19 +91,23 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
 
             @Override
             public void afterTextChanged(Editable s) {
+                if (s.length() > 5){
```

Comment:
You don't need to write a lot of code to do that - you can set the `android:maxLength` property on the EditText widget (in the xml file) and that will limit the length